backend/dgraph/src/drug_interaction/interactions.rs

```diff
@@ -0,0 +1,118 @@
+use gql_client::GraphQLError;
+use serde::Serialize;
+
+use crate::DgraphClient;
+use crate::DrugInteractionsData;
+
+#[derive(Serialize, Debug, Clone)]
+pub struct DrugInteractionFilter {
+    pub name: Option<String>,
+}
+
+pub async fn interactions(
+    client: &DgraphClient,
+    filter: Option<DrugInteractionFilter>,
```

Comment:
NIT but seeing as we've named this `variables` everywhere else, I saw filter and expected that it would be the entire filter struct to pass into the query i.e. `{name: {eq: $name}}`

Reply:
I had thought I might make it a more fully featured filter, but seeing as I haven't I agree...

Reply:
[99309fd](https://github.com/msupply-foundation/unified-codes/pull/574/commits/99309fd5e739ff8298d89547ee07b00936ac4172)